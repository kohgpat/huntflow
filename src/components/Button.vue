<template>
  <button v-bind:class="getClass">
    {{ text }}
  </button>
</template>

<script>
export default {
  name: 'Button',
  props: {
    text: String
  },
  computed: {
    getClass: () => ({
      "button": true,
      "button--default": true
    })
  }
}
</script>

<style scoped>
  .button {
    padding: 9px 15px 8px;
    font-weight: 700;
    font-size: 11px;
    line-height: 100%;
    text-transform: uppercase;
    background-color: #f1f1f1;
    border: 1px solid #f1f1f1;
    border-radius: 3px;
    color: #000;
  }

  .button--default {
    background-color: #7AC016;
    border: 1px solid #7AC016;
    color: #fff;
  }
</style>