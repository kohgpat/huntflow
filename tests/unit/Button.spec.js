import { shallowMount } from "@vue/test-utils";
import Button from "@/components/Button.vue";

describe("Button", () => {
  it("should match the snapshot", () => {
    const defaultButton = shallowMount(Button);
    const primaryButton = shallowMount(Button, {
      propsData: { variant: "primary" }
    });
    const successButton = shallowMount(Button, {
      propsData: { variant: "success" }
    });
    const dangerButton = shallowMount(Button, {
      propsData: { variant: "danger" }
    });

    expect(defaultButton).toMatchSnapshot();
    expect(successButton).toMatchSnapshot();
    expect(primaryButton).toMatchSnapshot();
    expect(dangerButton).toMatchSnapshot();
  });
});
