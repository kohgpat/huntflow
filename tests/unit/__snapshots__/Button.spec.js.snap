// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `<button class="button"></button>`;

exports[`Button should match the snapshot 2`] = `<button class="button button--success"></button>`;

exports[`Button should match the snapshot 3`] = `<button class="button button--primary"></button>`;

exports[`Button should match the snapshot 4`] = `<button class="button button--danger"></button>`;
